consensus/src/quorum_store/batch_store.rs

```diff
@@ -266,69 +251,39 @@ impl<T: QuorumStoreSender + Clone + Send + Sync + 'static> BatchStore<T> {
         self.expirations
             .lock()
             .unwrap()
-            .add_item(digest, expiration_round);
+            .add_item(digest, expiration_time);
         Ok(true)
     }
 
     pub(crate) fn save(&self, digest: HashValue, value: PersistedValue) -> anyhow::Result<bool> {
-        let expiration = value.expiration();
-        if expiration.epoch() == self.epoch() {
-            // record the round gaps
-            if expiration.round() > self.last_certified_round() {
-                counters::GAP_BETWEEN_BATCH_EXPIRATION_AND_LAST_CERTIFIED_ROUND_HIGHER
-                    .observe((expiration.round() - self.last_certified_round()) as f64);
-            }
-            if expiration.round() < self.last_certified_round() {
-                counters::GAP_BETWEEN_BATCH_EXPIRATION_AND_LAST_CERTIFIED_ROUND_LOWER
-                    .observe((self.last_certified_round() - expiration.round()) as f64);
-            }
-
-            if expiration.round() + self.batch_expiry_round_gap_behind_latest_certified
-                >= self.last_certified_round()
-                && expiration.round()
-                    <= self.last_certified_round()
-                        + self.batch_expiry_round_gap_beyond_latest_certified
-            {
-                fail_point!("quorum_store::save", |_| {
-                    // Skip caching and storing value to the db
-                    Ok(false)
-                });
-
-                return self.insert_to_cache(digest, value);
-            }
+        let last_certified_time = self.last_certified_time();
```

Comment:
Do we want to add a similar counter here to see the timestamp gap between the batch and the receiver?